### PR TITLE
Adding functions for Pixbuf scaling

### PR DESCRIPTION
**Pull request by jashephe**
Added functions for scaling `Pixbuf`s as described in "[Scaling](https://developer.gnome.org/gdk-pixbuf/unstable/gdk-pixbuf-Scaling.html)".

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -139,7 +139,8 @@ pub use gdk_ffi::enums::{
     WindowWindowClass,
     Gravity,
     DragAction,
-    DragProtocol
+    DragProtocol,
```
Comment (gkoz) [inline src/lib.rs]: This is why trailing commas are recommended so adding or removing an element doesn't touch other lines.

File: src/pixbuf/mod.rs [hunk 2]
```diff
@@ -181,6 +183,66 @@ impl Pixbuf {
...
+    /// scaling down `Nearest` is usually unusably ugly). The default
+    /// `interp_type` should be `Bilinear` which offers reasonable quality and
+    /// speed.
+    ///
+    /// You can scale a sub-portion of `src` by creating a sub-pixbuf pointing
+    /// into src ; see `new_subpixbuf()`.
+    ///
+    /// For more complicated scaling/compositing see `scale()` and
+    /// `composite()`.
+    pub fn scale_simple(&self, dest_width: i32, dest_height: i32, interp_type: InterpType) -> Option<Pixbuf> {
```
Comment (gkoz) [inline src/pixbuf/mod.rs]: I believe we've settled on using `Result` for failures. At least that's what `new` does.
We could rethink this particular failure mode and just panic on the OOM failures.

File: src/pixbuf/mod.rs [hunk 3]
```diff
@@ -181,6 +183,66 @@ impl Pixbuf {
...
+        unsafe {
+            from_glib_none(ffi::gdk_pixbuf_scale_simple(self.to_glib_none().0, dest_width, dest_height, interp_type))
```
Comment (gkoz) [inline src/pixbuf/mod.rs]: The sys crates are a lost cause but otherwise let's try to follow the 100-column rule.

**Comment by GuillaumeGomez** [comment]
@gkoz: Do you want it to be merged like that ? The only real problem is the `Option` stuff but since we didn't start to change the stuff, does it matter for now ?

**Comment by gkoz** [comment]
The Pixbuf part seems to have been converted to Results. I guess we could do it but pushing to crates like this would commit the API until the next push when we would break it once again.

**Comment by GuillaumeGomez** [comment]
I can wait, we're not in a hurry.

File: src/pixbuf/mod.rs [hunk 4]
```diff
@@ -181,6 +183,77 @@ impl Pixbuf {
...
+        unsafe {
+            ffi::gdk_pixbuf_composite(self.to_glib_none().0, dest.to_glib_none().0, dest_x, dest_y,
+                                      dest_width, dest_height, offset_x, offset_y, scale_x,
+                                      scale_y, interp_type, overall_alpha);
+        }
+    }
+
+    /// Flips a pixbuf horizontally or vertically and returns the result in a
+    /// new pixbuf, or `None` if not enough memory could be allocated for it.
+    pub fn flip(&self, horizontal: bool) -> Option<Pixbuf> {
```
Comment (gkoz) [inline src/pixbuf/mod.rs]: This one seems to be missing the same `Result` treatment.

Comment (jashephe) [inline src/pixbuf/mod.rs]: Indeed.  Give me a sec.

**Comment by gkoz** [comment]
Looks good!

**Comment by GuillaumeGomez** [comment]
Perfect, I merge. Thanks for your work ! =D

**Pull request merged**